src/System.ComponentModel.Primitives/src/System/ComponentModel/CategoryAttribute.cs

```diff
@@ -0,0 +1,335 @@
+        /// </devdoc>
+        public static CategoryAttribute Layout
+        {
+            get
+            {
+                if (s_layout == null)
+                {
+                    s_layout = new CategoryAttribute(nameof(Layout));
+                }
+                return s_layout;
+            }
+        }
+
+        /// <devdoc>
+        ///    <para>Gets the mouse category attribute.</para>
+        /// </devdoc>
+        public static CategoryAttribute Mouse
+        {
+            get
+            {
+                if (s_mouse == null)
+                {
+                    s_mouse = new CategoryAttribute(nameof(Mouse));
+                }
+                return s_mouse;
+            }
+        }
+
+        /// <devdoc>
+        ///    <para> Gets the window style category 
+        ///       attribute.</para>
+        /// </devdoc>
+        public static CategoryAttribute WindowStyle
+        {
+            get
+            {
+                if (s_windowStyle == null)
+                {
+                    s_windowStyle = new CategoryAttribute(nameof(WindowStyle));
+                }
+                return s_windowStyle;
+            }
+        }
+
+        /// <devdoc>
+        /// <para>Initializes a new instance of the <see cref='System.ComponentModel.CategoryAttribute'/> 
+        /// class with the default category.</para>
+        /// </devdoc>
+        public CategoryAttribute() : this(nameof(Default))
+        {
+        }
+
+        /// <devdoc>
+        /// <para>Initializes a new instance of the <see cref='System.ComponentModel.CategoryAttribute'/> class with
+        ///    the specified category name.</para>
+        /// </devdoc>
+        public CategoryAttribute(string category)
+        {
+            _categoryValue = category;
+            _localized = false;
+        }
+
+        /// <devdoc>
+        ///    <para>Gets the name of the category for the property or event 
+        ///       that this attribute is bound to.</para>
+        /// </devdoc>
+        public string Category
+        {
+            get
+            {
+                if (!_localized)
+                {
+                    _localized = true;
+                    string localizedValue = GetLocalizedString(_categoryValue);
+                    if (localizedValue != null)
+                    {
+                        _categoryValue = localizedValue;
+                    }
+                }
+                return _categoryValue;
+            }
+        }
+
+        /// <devdoc>
+        /// </devdoc>
+        /// <devdoc>
+        /// </devdoc>
+        /// <internalonly/>
+        /// <internalonly/>
+        public override bool Equals(object obj)
+        {
+            if (obj == this)
+            {
+                return true;
+            }
+
+            if (obj is CategoryAttribute)
+            {
+                return Category.Equals(((CategoryAttribute)obj).Category);
+            }
+            return false;
```

Comment:
This would be a bit better as the following (which also matches the pattern used earlier in BrowsableAttribute):
```C#
CategoryAttribute other = obj as CategoryAttribute;
return other != null && Category.Equals(other.Category);
```